doc/v5.0/man3/MPI_Comm_disconnect.3
.\" Automatically generated by Pandoc 2.14.2
.\"
.TH "MPI_Comm_disconnect" "3" "" "2021-08-30" "Open MPI"
.hy
.SH Name
.PP
\f[C]MPI_Comm_disconnect\f[R] - Deallocates communicator object and sets
handle to \f[C]MPI_COMM_NULL.\f[R]
.SH Syntax
.SS C Syntax
.IP
.nf
\f[C]
#include <mpi.h>

int MPI_Comm_disconnect(MPI_Comm *comm)
\f[R]
.fi
.SS Fortran Syntax
.IP
.nf
\f[C]
USE MPI
! or the older form: INCLUDE \[aq]mpif.h\[aq]

MPI_COMM_DISCONNECT(COMM, IERROR)
    INTEGER COMM, IERROR 
\f[R]
.fi
.SS Fortran 2008 Syntax
.IP
.nf
\f[C]
USE mpi_f08

MPI_Comm_disconnect(comm, ierror)
    TYPE(MPI_Comm), INTENT(INOUT) :: comm
    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
\f[R]
.fi
.SS Input/Output Parameter
.IP \[bu] 2
\f[C]comm\f[R] : Communicator (handle).
.SH Output Parameter
.IP \[bu] 2
\f[C]IERROR\f[R] : Fortran only: Error status (integer).
.SH Description
.PP
\f[C]MPI_Comm_disconnect\f[R] waits for all pending communication on
\f[C]comm\f[R] to complete internally, deallocates the communicator
object, and sets the handle to \f[C]MPI_COMM_NULL\f[R].
It is a collective operation.
It may not be called with the communicator \f[C]MPI_COMM_WORLD\f[R] or
\f[C]MPI_COMM_SELF.\f[R] \f[C]MPI_Comm_disconnect\f[R] may be called
only if all communication is complete and matched, so that buffered data
can be delivered to its destination.
This requirement is the same as for \f[C]MPI_Finalize.\f[R]
\f[C]MPI_Comm_disconnect\f[R] has the same action as
\f[C]MPI_Comm_free\f[R], except that it waits for pending communication
to finish internally and enables the guarantee about the behavior of
disconnected processes.
.SH Notes
.PP
To disconnect two processes you may need to call
\f[C]MPI_Comm_disconnect,\f[R] \f[C]MPI_Win_free\f[R], and
\f[C]MPI_File_close\f[R] to remove all communication paths between the
two processes.
Note that it may be necessary to disconnect several communicators (or to
free several windows or files) before two processes are completely
independent.
.SH Errors
.PP
Almost all MPI routines return an error value; C routines as the value
of the function and Fortran routines in the last argument.
Before the error value is returned, the current MPI error handler is
called.
By default, this error handler aborts the MPI job, except for I/O
function errors.
The error handler may be changed with \f[C]MPI_Comm_set_errhandler\f[R];
the predefined error handler \f[C]MPI_ERRORS_RETURN\f[R] may be used to
cause error values to be returned.
Note that MPI does not guarantee that an MPI program can continue past
an error.
.SH See Also
.PP
\f[C]MPI_Comm_connect\f[R](3) \f[C]MPI_Comm_accept\f[R](3)


doc/v1.4/man3/MPI_Comm_disconnect.3.php
<?php
$topdir = "../../..";
$title = "MPI_Comm_disconnect(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_COMM_DISCONNECT(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Comm_disconnect</b> - Deallocates communicator object and sets handle
to MPI_COMM_NULL.
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Comm_disconnect(MPI_Comm *comm)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_COMM_DISCONNECT(COMM, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;COMM, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
void MPI::Comm::Disconnect()
</pre>
<h2><a name='sect5' href='#toc5'>Input/Output Parameter</a></h2>

<dl>

<dt>comm </dt>
<dd>Communicator (handle).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Output Parameter</a></h2>

<dl>

<dt>IERROR
</dt>
<dd>Fortran only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Description</a></h2>
MPI_Comm_disconnect waits
for all pending communication on <i>comm</i> to complete internally, deallocates
the communicator object, and sets the handle to MPI_COMM_NULL. It is a collective
operation.  <p>
It may not be called with the communicator MPI_COMM_WORLD or
MPI_COMM_SELF. <p>
MPI_Comm_disconnect may be called only if all communication
is complete and matched, so that buffered data can be delivered to its
destination. This requirement is the same as for <a href="../man3/MPI_Finalize.3.php">MPI_Finalize</a>.  <p>
MPI_Comm_disconnect
has the same action as <a href="../man3/MPI_Comm_free.3.php">MPI_Comm_free</a>, except that it waits for pending
communication to finish internally and enables the guarantee about the
behavior of disconnected processes.
<p>
<h2><a name='sect8' href='#toc8'>Notes</a></h2>
To disconnect two processes you
may need to call MPI_Comm_disconnect, <a href="../man3/MPI_Win_free.3.php">MPI_Win_free</a>, and <a href="../man3/MPI_File_close.3.php">MPI_File_close</a>
to remove all communication paths between the two processes. Note that it
may be necessary to disconnect several communicators (or to free several
windows or files) before two processes are completely independent.
<p>
<h2><a name='sect9' href='#toc9'>Errors</a></h2>
Almost
all MPI routines return an error value; C routines as the value of the
function and Fortran routines in the last argument. C++ functions do not
return errors. If the default error handler is set to MPI::ERRORS_THROW_EXCEPTIONS,
then on error the C++ exception mechanism will be used to throw an MPI:Exception
object. <p>
Before the error value is returned, the current MPI error handler
is called. By default, this error handler aborts the MPI job, except for
I/O function errors. The error handler may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>;
the predefined error handler MPI_ERRORS_RETURN may be used to cause error
values to be returned. Note that MPI does not guarantee that an MPI program
can continue past an error.
<p>
<h2><a name='sect10' href='#toc10'>See Also</a></h2>
<p>
<a href="../man3/MPI_Comm_connect.3.php">MPI_Comm_connect</a> <br>
<a href="../man3/MPI_Comm_accept.3.php">MPI_Comm_accept</a> <br>
<p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>C++ Syntax</a></li>
<li><a name='toc5' href='#sect5'>Input/Output Parameter</a></li>
<li><a name='toc6' href='#sect6'>Output Parameter</a></li>
<li><a name='toc7' href='#sect7'>Description</a></li>
<li><a name='toc8' href='#sect8'>Notes</a></li>
<li><a name='toc9' href='#sect9'>Errors</a></li>
<li><a name='toc10' href='#sect10'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
